### PR TITLE
Geo/geotools itwin4 2

### PR DESCRIPTION
**Pull request by mdastous-bentley**

### CODE REVIEW

File: packages/itwin/geo-tools/src/GeoToolsUiItemsProvider.tsx [hunk 1]
```diff
@@ -2,28 +2,20 @@
...
-  CommonToolbarItem,
-  UiItemsProvider,
-} from "@itwin/appui-abstract";
-import {
-  StageUsage,
-  ToolbarOrientation,
-  ToolbarUsage,
-} from "@itwin/appui-abstract";
-import { ToolbarHelper } from "@itwin/appui-react";
+// eslint-disable-next-line @typescript-eslint/consistent-type-imports
```
Comment (aruniverse) [inline packages/itwin/geo-tools/src/GeoToolsUiItemsProvider.tsx:5]: why is this disabled? can just run `lint --fix`

File: common/changes/@itwin/geo-tools-react/geo-geotools-itwin4_2023-03-28-14-45.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/geo-tools-react",
+      "comment": "Migrated geo-tools to iTwin.js 4.0",
```
Comment (aruniverse) [inline common/changes/@itwin/geo-tools-react/geo-geotools-itwin4_2023-03-28-14-45.json:5]:
```suggestion
      "comment": "Migrated geo-tools to iTwin.js 3.7",
```

File: packages/itwin/geo-tools/src/GeoToolsUiItemsProvider.tsx (hunk 1 above)
Comment (mdastous-bentley) [reply packages/itwin/geo-tools/src/GeoToolsUiItemsProvider.tsx:5]: The fix normally be to use:
`import type UiItemsProvider from "@itwin/appui-react";`

When I use this, I get a compile error:
`src/GeoToolsUiItemsProvider.tsx(10,55): error TS2709: Cannot use namespace 'UiItemsProvider' as a type.`

**Comment by mdastous-bentley** [comment]
@aruniverse Are you able to run the tests?  In the acutal configuration I get:
`Error: Multiple @itwin/core-frontend imports detected!`

That started to appear when downgrading itwin.js from 4.0 to 3.7

File: packages/itwin/geo-tools/src/GeoToolsUiItemsProvider.tsx (hunk 1 above)
Comment (jason-crow) [reply packages/itwin/geo-tools/src/GeoToolsUiItemsProvider.tsx:5]: > The fix normally be to use: `import type UiItemsProvider from "@itwin/appui-react";`
> 
> When I use this, I get a compile error: `src/GeoToolsUiItemsProvider.tsx(10,55): error TS2709: Cannot use namespace 'UiItemsProvider' as a type.`

import { type UiItemsProvider } from "@itwin/appui-react" / import type { UiItemsProvider } from "@itwin/appui-react"

Comment (mdastous-bentley) [reply packages/itwin/geo-tools/src/GeoToolsUiItemsProvider.tsx]: Thanks Jason, I was missing only the curly brackets... the lint message could be improved here.

**Comment by aruniverse** [comment]
> @aruniverse Tests still refuse to run, do dependencies versions look good?

@Jake-Screen please work with Michel on solving the duplicate core-frontends

**Comment by Jake-Screen** [comment]
> > @aruniverse Tests still refuse to run, do dependencies versions look good?
> 
> @Jake-Screen please work with Michel on solving the duplicate core-frontends

Sounds good, will do

**Comment by mdastous-bentley** [comment]
@aruniverse @Jake-Screen All problems resolved!  Please approve

**Review by ben-polinsky — APPROVED** [review]

File: packages/itwin/grouping-mapping-widget/src/widget/components/CalculatedPropertyTable.tsx
Comment (aruniverse) [inline packages/itwin/grouping-mapping-widget/src/widget/components/CalculatedPropertyTable.tsx:1]: revert this file

File: packages/itwin/geo-tools/src/GeoToolsUiItemsProvider.tsx [hunk 3]
```diff
@@ -2,28 +2,20 @@
...
   public readonly id = "GeoToolsAddressSearchProvider";
 
-  public provideToolbarButtonItems(
+  public  provideToolbarItems(
     _stageId: string,
     stageUsage: string,
     toolbarUsage: ToolbarUsage,
     toolbarOrientation: ToolbarOrientation
-  ): CommonToolbarItem[] {
+  ) {
```
Comment (aruniverse) [inline packages/itwin/geo-tools/src/GeoToolsUiItemsProvider.tsx:18]: return type?

File: common/changes/@itwin/geo-tools-react/geo-geotools-itwin4_2023-03-28-14-45.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/geo-tools-react",
+      "comment": "Migrated geo-tools to iTwin.js 3.7",
+      "type": "minor"
```
Comment (aruniverse) [inline common/changes/@itwin/geo-tools-react/geo-geotools-itwin4_2023-03-28-14-45.json:6]:
```suggestion
      "type": "none"
```

**Review by aruniverse — COMMENTED** [review]
@Jake-Screen can you point @mdastous-bentley to make the same changes needed to set up a pre-release like we did for measure-tools and the tree-widget/property-grid

**Comment by Jake-Screen** [comment]
> @Jake-Screen can you point @mdastous-bentley to make the same changes needed to set up a pre-release like we did for measure-tools and the tree-widget/property-grid

Here is a link to how tree-widgets was set to pre-release @mdastous-bentley https://github.com/iTwin/viewer-components-react/commit/33fbd7dd303c8f0a2033b877b58a6fa6baac01f3

File: packages/itwin/geo-tools/src/GeoToolsUiItemsProvider.tsx (hunk 3 above)
Comment (mdastous-bentley) [reply packages/itwin/geo-tools/src/GeoToolsUiItemsProvider.tsx]: fixed

File: packages/itwin/grouping-mapping-widget/src/widget/components/CalculatedPropertyTable.tsx
Comment (mdastous-bentley) [reply packages/itwin/grouping-mapping-widget/src/widget/components/CalculatedPropertyTable.tsx:1]: done

**Review by aruniverse — APPROVED** [review]

**Pull request merged**